paddle/fluid/ir_adaptor/translator/op_translator.cc

```diff
@@ -2470,6 +2470,45 @@ struct ShareBufferOpTranscriber : public OpTranscriber {
   }
 };
 
+struct RandIntOpTranscriber : public OpTranscriber {
+  std::tuple<OpOutputTypeList, OpOutputMapping> GenerateOperationOutput(
+      pir::IrContext* ctx,
+      const OpDesc& op_desc,
+      const OpOutputInfoList& output_infos) {
+    OpOutputMapping arg_to_idx;
+    OpOutputTypeList op_output_types = {};
+
+    auto& type_translator = TypeTranslator::instance();
+
+    const BlockDesc* block = op_desc.Block();
+    std::string legacy_output_name = "Out";
+    const auto& legacy_output_vars = op_desc.Output(legacy_output_name);
+    auto& var_name = legacy_output_vars[0];
+    VarDesc* var = block->FindVarRecursive(var_name);
+    IR_ENFORCE(var != nullptr,
+               "[op:%s] Output %s should not be null",
+               op_desc.Type(),
+               var_name);
+    int dtype_num = PADDLE_GET_CONST(int, op_desc.GetAttr("dtype"));
+
+    paddle::framework::proto::VarType::Type var_type =
+        static_cast<paddle::framework::proto::VarType::Type>(dtype_num);
+
+    pir::Type dtype = type_translator.operator[](var_type)(ctx, *var);
```

Comment:
```suggestion
    pir::Type dtype = type_translator[var_type](ctx, *var);
```
这行应该可以这样？

另外，输出类型的翻译过程能否优化为
```
pir::Type translated_var_type = type_translator[VarType::LOD_TENSOR](ctx, *var);
```

Reply:
已修改为
```C++
 pir::Type dtype = type_translator[var_type](ctx, *var);
```
这里主要是要翻译的LOD_TENSOR的dtype类型会出现RAW类型，直接调用type_translator[VarType::LOD_TENSOR](ctx, *var)的话，会调用对应的函数翻译RAW类型，但是目前不支持RAW类型的翻译，所以这里仿照InferMeta的逻辑，将attribute中的dtype指定给Out